src/Symfony/Component/Messenger/Bridge/AmazonSqs/Middleware/SupportsMessageDeduplicationInterface.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Bridge\AmazonSqs\Middleware;
+
+/**
+ * @see https://docs.aws.amazon.com/en_gb/AWSSimpleQueueService/latest/SQSDeveloperGuide/using-messagededuplicationid-property.html
+ */
+interface SupportsMessageDeduplicationInterface
+{
+    /**
+     * The max length of MessageDeduplicationId is 128 characters.
+     * Valid values: alphanumeric characters and punctuation (!"#$%&'()*+,-./:;<=>?@[]^_`{|}~).
+     */
+    public function messageDeduplicationId(): string;
```

Comment:
should we make the method nullable? (same question for the other interface)
it might be more flexible

Reply:
That would indeed be more flexible/dynamic. But it what scenario would it be used?

Reply:
E.g. allowing a child class to override the parent and return null

Reply:
Yeah.. You are correct. But I am more "that will never happen".

But I see no point it not allowing this because I think it would be unlikely. Let's be pragmatic and allow null values.